src/main/java/io/nats/client/impl/NatsMessageConsumerBase.java

```diff
@@ -101,21 +101,19 @@ public void stop() {
 
     @Override
     public void close() throws Exception {
-        lenientClose();
+        stopped.set(true);
+        shutdownSub();
     }
 
-    protected void finishAndClose() {
-        if (pmm != null) {
-            pmm.shutdownHeartbeatTimer();
-        }
+    protected void fullClose() {
```

Comment:
This consume is completely done. Stopped, finished, caput!
1. From fetch consumer since fetch is not endless, when the fetch has been satisfied or there is a heartbeat error which means there aren't any more messages coming.
2. From NatsMessageConsumer (endless consumes) only when the user has told the consumer to stop() and the latest pull is complete or there was a heartbeat error